### PR TITLE
[SOAR-16852] Proofpoint Initial Lookback

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Update initial lookback to 7 days
  - Update SDK

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/proofpoint_tap/help.md [hunk 1]
```diff
@@ -74,243 +75,157 @@ Example input:
...
-              "os": "win",
-              "version": "win10"
-            }
-          ]
-        }
-      ]
-    }
-  ]
+  "generated": "2021-06-27 19:58:04.283000+00:00",
+  "reports": []
```
Comment (cmcnally-r7) [inline plugins/proofpoint_tap/help.md:87]: can we restore this

File: plugins/proofpoint_tap/help.md [hunk 2]
```diff
@@ -74,243 +75,157 @@ Example input:
...
 ```
 {
-  "urls": [
-    {
-      "encodedUrl": "https://urldefense.proofpoint.com/v1/url?u=http://www.example.com/&amp;k=oIvRg1%2BdGAgOoM1BIlLLqw%3D%3D%0A&amp;r=IKM5u8%2B%2F%2Fi8EBhWOS%2BqGbTqCC%2BrMqWI%2FVfEAEsQO%2F0Y%3D%0A&amp;m=Ww6iaHO73mDQpPQwOwfLfN8WMapqHyvtu8jM8SjqmVQ%3D%0A&amp;s=d3583cfa53dade97025bc6274c6c8951dc29fe0f38830cf8e5a447723b9f1c9a\"",
-      "decodedUrl": "http://www.example.com/",
-      "success": true
-    }
-  ]
+  "results": {}
```
Comment (cmcnally-r7) [inline plugins/proofpoint_tap/help.md:126]: is there an example we can put here too

File: plugins/proofpoint_tap/help.md [hunk 3]
```diff
@@ -74,243 +75,157 @@ Example input:
...
-      "identity": {
-        "emails": [
-          "user@example.com"
-        ],
-        "guid": "9de5069c-5afe-602b-2ea0-a04b66beb2c0",
-        "vip": false
-      }
-    }
-  ]
+  "results": {}
```
Comment (cmcnally-r7) [inline plugins/proofpoint_tap/help.md:164]: and here

File: plugins/proofpoint_tap/help.md [hunk 4]
```diff
@@ -319,858 +234,501 @@ Example input:
...
-        "senderIP": "198.51.100.1",
-        "id": "9de5069c-5afe-602b-2ea0-a04b66beb2c0"
-      }
-    ],
-    "messagesDelivered": [],
-    "queryEndTime": "2021-04-21T12:00:00Z"
-  }
+  "interval": "2023-03-07T06:01:00Z/2023-06-05T06:01:00Z",
+  "totalTopClickers": 2,
+  "users": []
```
Comment (cmcnally-r7) [inline plugins/proofpoint_tap/help.md:319]: and here and so on for all empty `[]` and `{}` values

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

File: plugins/proofpoint_tap/Dockerfile [hunk 5]
```diff
@@ -17,10 +17,4 @@ RUN python setup.py build && python setup.py install
 # User to run plugin code. The two supported users are: root, nobody
 USER nobody
 
-# PLGN-701: POC how we can inject an env var into the container to specify a new date
```
Comment (cmcnally-r7) [inline plugins/proofpoint_tap/Dockerfile:20]: gonna approve but double check we don't need this comment anymore

**Review by cmcnally-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/Dockerfile (hunk 5 above)
Comment (ablakley-r7) [reply plugins/proofpoint_tap/Dockerfile:20]: This has been replaced by backfill logic. Comment moved to wiki for posterity.

**Review by ekelly-r7 — APPROVED** [review]

**Pull request merged**